src/lauxlib.js

```diff
@@ -783,6 +787,70 @@ if (!WEB) {
     module.exports.luaL_dofile    = luaL_dofile;
     module.exports.luaL_loadfilex = luaL_loadfilex;
     module.exports.luaL_loadfile  = luaL_loadfile;
+} else {
+    const getF = function(L, ud) {
+        let lf = ud;
+        let f = lf.f;
+        lf.f = null;
+        return f;
+    };
+
+    getc = function(lf) {
+        return lf.pos < lf.f.length ? lf.f[lf.pos++] : null;
+    };
+
+    const luaL_loadfilex = function(L, filename, mode) {
+        let lf = new LoadF();
+        let fnameindex = lua.lua_gettop(L) + 1;  /* index of filename on the stack */
+        if (filename === null) {
+            throw new Error(`Can't read stdin in the browser`);
+        } else {
+            let jsfilename = lua.to_jsstring(filename);
+            lua.lua_pushliteral(L, `@${jsfilename}`);
+
+            let xhr = new XMLHttpRequest();
+            xhr.open("GET", jsfilename, false);
+            //xhr.responseType = "arraybuffer";
+            xhr.send();
+
+            if (xhr.status === 200) {
+                /* TODO: Synchronous xhr alway return a js string */
+                lf.f = lua.to_luastring(xhr.response);
```

Comment:
No `lf.buff` is used by getF (node version) to read a file by chunk of 1024 bytes, but the browser version reads the whole file in one go.